README.rst
Python Client for App Engine Logging Protos API
===============================================

|stable| |pypi| |versions|

`App Engine Logging Protos API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-appengine-logging.svg
   :target: https://pypi.org/project/google-cloud-appengine-logging/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-appengine-logging.svg
   :target: https://pypi.org/project/google-cloud-appengine-logging/
.. _App Engine Logging Protos API: https://cloud.google.com/logging/docs/reference/v2/rpc/google.appengine.logging.v1
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference//latest
.. _Product Documentation:  https://cloud.google.com/logging/docs/reference/v2/rpc/google.appengine.logging.v1

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the App Engine Logging Protos API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the App Engine Logging Protos API.:  https://cloud.google.com/logging/docs/reference/v2/rpc/google.appengine.logging.v1
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current `active`_ and `maintenance`_ versions of
Python.

Python >= 3.7

.. _active: https://devguide.python.org/devcycle/#in-development-main-branch
.. _maintenance: https://devguide.python.org/devcycle/#maintenance-branches

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an `end-of-life`_
version of Python, we recommend that you update as soon as possible to an actively supported version.

.. _end-of-life: https://devguide.python.org/devcycle/#end-of-life-branches

Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-appengine-logging


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-appengine-logging

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for App Engine Logging Protos API
   to see other available methods on the client.
-  Read the `App Engine Logging Protos API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _App Engine Logging Protos API Product documentation:  https://cloud.google.com/logging/docs/reference/v2/rpc/google.appengine.logging.v1
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [0.1.4](https://www.github.com/googleapis/python-appengine-logging/compare/v0.1.3...v0.1.4) (2021-07-29)


### Features

* add Samples section to CONTRIBUTING.rst ([#19](https://www.github.com/googleapis/python-appengine-logging/issues/19)) ([8854cbe](https://www.github.com/googleapis/python-appengine-logging/commit/8854cbe02c233901bf4601a99cdfbd9826b616bf))


### Miscellaneous Chores

* release as 0.1.4 ([#24](https://www.github.com/googleapis/python-appengine-logging/issues/24)) ([5aa924e](https://www.github.com/googleapis/python-appengine-logging/commit/5aa924e0a24a94cf259168cc1dd5aa65dc0f40a9))

### [0.1.3](https://www.github.com/googleapis/python-appengine-logging/compare/v0.1.2...v0.1.3) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#18](https://www.github.com/googleapis/python-appengine-logging/issues/18)) ([6f49270](https://www.github.com/googleapis/python-appengine-logging/commit/6f492709a55926c7b993da7bb26469fc0bf79128))

### [0.1.2](https://www.github.com/googleapis/python-appengine-logging/compare/v0.1.1...v0.1.2) (2021-07-14)


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-appengine-logging/issues/1127)) ([#8](https://www.github.com/googleapis/python-appengine-logging/issues/8)) ([f1bd0a3](https://www.github.com/googleapis/python-appengine-logging/commit/f1bd0a3d2d068fae0c6d9c167a25908b6f808997)), closes [#1126](https://www.github.com/googleapis/python-appengine-logging/issues/1126)

### [0.1.1](https://www.github.com/googleapis/python-appengine-logging/compare/v0.1.0...v0.1.1) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#5](https://www.github.com/googleapis/python-appengine-logging/issues/5)) ([7f65db0](https://www.github.com/googleapis/python-appengine-logging/commit/7f65db00cd1196d2fc2d7ec2197a594d6b56a568))

## 0.1.0 (2021-06-02)


### Features

* generate google.appengine.logging.v1.request_log ([e1c6c72](https://www.github.com/googleapis/python-appengine-logging/commit/e1c6c7218b098788b5639f09e173cbb3d01ee09b))


owl-bot-staging/v1/google/cloud/appengine_logging_v1/types/request_log.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import proto  # type: ignore

from google.logging.type import log_severity_pb2  # type: ignore
from google.protobuf import duration_pb2  # type: ignore
from google.protobuf import timestamp_pb2  # type: ignore


__protobuf__ = proto.module(
    package='google.appengine.logging.v1',
    manifest={
        'LogLine',
        'SourceLocation',
        'SourceReference',
        'RequestLog',
    },
)


class LogLine(proto.Message):
    r"""Application log line emitted while processing a request.
    Attributes:
        time (google.protobuf.timestamp_pb2.Timestamp):
            Approximate time when this log entry was
            made.
        severity (google.logging.type.log_severity_pb2.LogSeverity):
            Severity of this log entry.
        log_message (str):
            App-provided log message.
        source_location (google.cloud.appengine_logging_v1.types.SourceLocation):
            Where in the source code this log message was
            written.
    """

    time = proto.Field(
        proto.MESSAGE,
        number=1,
        message=timestamp_pb2.Timestamp,
    )
    severity = proto.Field(
        proto.ENUM,
        number=2,
        enum=log_severity_pb2.LogSeverity,
    )
    log_message = proto.Field(
        proto.STRING,
        number=3,
    )
    source_location = proto.Field(
        proto.MESSAGE,
        number=4,
        message='SourceLocation',
    )


class SourceLocation(proto.Message):
    r"""Specifies a location in a source code file.
    Attributes:
        file (str):
            Source file name. Depending on the runtime
            environment, this might be a simple name or a
            fully-qualified name.
        line (int):
            Line within the source file.
        function_name (str):
            Human-readable name of the function or method being invoked,
            with optional context such as the class or package name.
            This information is used in contexts such as the logs
            viewer, where a file and line number are less meaningful.
            The format can vary by language. For example:
            ``qual.if.ied.Class.method`` (Java), ``dir/package.func``
            (Go), ``function`` (Python).
    """

    file = proto.Field(
        proto.STRING,
        number=1,
    )
    line = proto.Field(
        proto.INT64,
        number=2,
    )
    function_name = proto.Field(
        proto.STRING,
        number=3,
    )


class SourceReference(proto.Message):
    r"""A reference to a particular snapshot of the source tree used
    to build and deploy an application.

    Attributes:
        repository (str):
            Optional. A URI string identifying the
            repository. Example:
            "https://github.com/GoogleCloudPlatform/kubernetes.git".
        revision_id (str):
            The canonical and persistent identifier of
            the deployed revision. Example (git):
            "0035781c50ec7aa23385dc841529ce8a4b70db1b".
    """

    repository = proto.Field(
        proto.STRING,
        number=1,
    )
    revision_id = proto.Field(
        proto.STRING,
        number=2,
    )


class RequestLog(proto.Message):
    r"""Complete log information about a single HTTP request to an
    App Engine application.

    Attributes:
        app_id (str):
            Application that handled this request.
        module_id (str):
            Module of the application that handled this
            request.
        version_id (str):
            Version of the application that handled this
            request.
        request_id (str):
            Globally unique identifier for a request,
            which is based on the request start time.
            Request IDs for requests which started later
            will compare greater as strings than those for
            requests which started earlier.
        ip (str):
            Origin IP address.
        start_time (google.protobuf.timestamp_pb2.Timestamp):
            Time when the request started.
        end_time (google.protobuf.timestamp_pb2.Timestamp):
            Time when the request finished.
        latency (google.protobuf.duration_pb2.Duration):
            Latency of the request.
        mega_cycles (int):
            Number of CPU megacycles used to process
            request.
        method (str):
            Request method. Example: ``"GET"``, ``"HEAD"``, ``"PUT"``,
            ``"POST"``, ``"DELETE"``.
        resource (str):
            Contains the path and query portion of the URL that was
            requested. For example, if the URL was
            "http://example.com/app?name=val", the resource would be
            "/app?name=val". The fragment identifier, which is
            identified by the ``#`` character, is not included.
        http_version (str):
            HTTP version of request. Example: ``"HTTP/1.1"``.
        status (int):
            HTTP response status code. Example: 200, 404.
        response_size (int):
            Size in bytes sent back to client by request.
        referrer (str):
            Referrer URL of request.
        user_agent (str):
            User agent that made the request.
        nickname (str):
            The logged-in user who made the request.

            Most likely, this is the part of the user's email before the
            ``@`` sign. The field value is the same for different
            requests from the same user, but different users can have
            similar names. This information is also available to the
            application via the App Engine Users API.

            This field will be populated starting with App Engine
            1.9.21.
        url_map_entry (str):
            File or class that handled the request.
        host (str):
            Internet host and port number of the resource
            being requested.
        cost (float):
            An indication of the relative cost of serving
            this request.
        task_queue_name (str):
            Queue name of the request, in the case of an
            offline request.
        task_name (str):
            Task name of the request, in the case of an
            offline request.
        was_loading_request (bool):
            Whether this was a loading request for the
            instance.
        pending_time (google.protobuf.duration_pb2.Duration):
            Time this request spent in the pending
            request queue.
        instance_index (int):
            If the instance processing this request
            belongs to a manually scaled module, then this
            is the 0-based index of the instance. Otherwise,
            this value is -1.
        finished (bool):
            Whether this request is finished or active.
        first (bool):
            Whether this is the first ``RequestLog`` entry for this
            request. If an active request has several ``RequestLog``
            entries written to Stackdriver Logging, then this field will
            be set for one of them.
        instance_id (str):
            An identifier for the instance that handled
            the request.
        line (Sequence[google.cloud.appengine_logging_v1.types.LogLine]):
            A list of log lines emitted by the
            application while serving this request.
        app_engine_release (str):
            App Engine release version.
        trace_id (str):
            Stackdriver Trace identifier for this
            request.
        trace_sampled (bool):
            If true, the value in the 'trace_id' field was sampled for
            storage in a trace backend.
        source_reference (Sequence[google.cloud.appengine_logging_v1.types.SourceReference]):
            Source code for the application that handled
            this request. There can be more than one source
            reference per deployed application if source
            code is distributed among multiple repositories.
    """

    app_id = proto.Field(
        proto.STRING,
        number=1,
    )
    module_id = proto.Field(
        proto.STRING,
        number=37,
    )
    version_id = proto.Field(
        proto.STRING,
        number=2,
    )
    request_id = proto.Field(
        proto.STRING,
        number=3,
    )
    ip = proto.Field(
        proto.STRING,
        number=4,
    )
    start_time = proto.Field(
        proto.MESSAGE,
        number=6,
        message=timestamp_pb2.Timestamp,
    )
    end_time = proto.Field(
        proto.MESSAGE,
        number=7,
        message=timestamp_pb2.Timestamp,
    )
    latency = proto.Field(
        proto.MESSAGE,
        number=8,
        message=duration_pb2.Duration,
    )
    mega_cycles = proto.Field(
        proto.INT64,
        number=9,
    )
    method = proto.Field(
        proto.STRING,
        number=10,
    )
    resource = proto.Field(
        proto.STRING,
        number=11,
    )
    http_version = proto.Field(
        proto.STRING,
        number=12,
    )
    status = proto.Field(
        proto.INT32,
        number=13,
    )
    response_size = proto.Field(
        proto.INT64,
        number=14,
    )
    referrer = proto.Field(
        proto.STRING,
        number=15,
    )
    user_agent = proto.Field(
        proto.STRING,
        number=16,
    )
    nickname = proto.Field(
        proto.STRING,
        number=40,
    )
    url_map_entry = proto.Field(
        proto.STRING,
        number=17,
    )
    host = proto.Field(
        proto.STRING,
        number=20,
    )
    cost = proto.Field(
        proto.DOUBLE,
        number=21,
    )
    task_queue_name = proto.Field(
        proto.STRING,
        number=22,
    )
    task_name = proto.Field(
        proto.STRING,
        number=23,
    )
    was_loading_request = proto.Field(
        proto.BOOL,
        number=24,
    )
    pending_time = proto.Field(
        proto.MESSAGE,
        number=25,
        message=duration_pb2.Duration,
    )
    instance_index = proto.Field(
        proto.INT32,
        number=26,
    )
    finished = proto.Field(
        proto.BOOL,
        number=27,
    )
    first = proto.Field(
        proto.BOOL,
        number=42,
    )
    instance_id = proto.Field(
        proto.STRING,
        number=28,
    )
    line = proto.RepeatedField(
        proto.MESSAGE,
        number=29,
        message='LogLine',
    )
    app_engine_release = proto.Field(
        proto.STRING,
        number=38,
    )
    trace_id = proto.Field(
        proto.STRING,
        number=39,
    )
    trace_sampled = proto.Field(
        proto.BOOL,
        number=43,
    )
    source_reference = proto.RepeatedField(
        proto.MESSAGE,
        number=41,
        message='SourceReference',
    )


__all__ = tuple(sorted(__protobuf__.manifest))
